lldb-eval/parser.cc

```diff
@@ -2022,6 +2032,14 @@ bool Parser::ImplicitConversionIsAllowed(Type src, Type dst) {
     }
   }
 
+  if (dst.IsPointerType()) {
+    // Literal zero, `nullptr_t` and arrays can be implicitly converted to
+    // pointers.
+    if (is_src_literal_zero || src.IsNullPtrType() || src.IsArrayType()) {
```

Comment:
For the case when src is an array, should we check that the array element type is the same as the pointee type?

Reply:
Oops, good point, we definitely should!